pkg/rpc/context.go

```diff
@@ -2715,13 +2715,14 @@ func (rpcCtx *Context) loadOrCreateConnAttempt(
 		select {
 		case <-previousAttempt.initialHeartbeatDone:
 			// The connection attempt was completed, return the outcome of it.
-			err := previousAttempt.err.Load().(error)
```

Comment:
slight preference for just changing this to `err, _ := previousAttempt.err.Load().(error)`, but only slight. This could still panic if somehow err.Load() had a non-nil, non-error value